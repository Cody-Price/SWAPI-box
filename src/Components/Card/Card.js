import React from 'react';
import './Card.scss';

const Card = ({card}) => {
  let displayedCard;
  if (Object.keys(card)[2] === 'species') {
    displayedCard = (
      <div>
        <h3>{card.name}</h3>
        <p>{card.homeworld}</p>
        <p>{card.species}</p>
        <p>{card.homeworldPop}</p>
        <button className='fav-btn' onClick={
          (e) => e.target.classList.toggle('active')
        }>Favorite</button>
      </div>
    )
  } else if (Object.keys(card)[1] === 'model') {
    displayedCard = (
      <div>
        <h3>{card.name}</h3>
        <p>{card.model}</p>
        <p>{card.class}</p>
        <p>{card.numberOfPassengers}</p>
        <button className='fav-btn' onClick={
          (e) => e.target.classList.toggle('active')
        }>Favorite</button>
      </div>
    )
  } else if (Object.keys(card)[1] === 'terrain') {
    displayedCard = (
      <div>
        <h3>{card.name}</h3>
        <p>{card.population}</p>
        <p>{card.climate}</p>
        <p>{card.terrain}</p>
          {
            card.residents.map(resident => {
              return <p key={resident}>{resident}</p>
            })
          }
        <button className='fav-btn' onClick={
          (e) => e.target.classList.toggle('active')
        }>Favorite</button>
      </div>
    )
  }
  return <div className='card'>{displayedCard}</div>
}

export default Card;