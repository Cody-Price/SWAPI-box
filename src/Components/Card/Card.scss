.card {
  display: inline-block;
  border: white 2px solid;
  color: white;
  height: 250px;
  width: 200px;
}

.active {
  background-color: yellow;
  color: black;
}