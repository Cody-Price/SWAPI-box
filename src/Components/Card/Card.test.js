import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {

  it('should match the snapshot if props includes "species"', () => {
    const card = {species: 'Human'}
    const wrapper = shallow(<Card card={card} />)
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot if props includes "model"', () => {
    const card = {model: 'x-wing'}
    const wrapper = shallow(<Card card={card} />);
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot if props includes "terrain"', () => {
    const card = {terrain: 'mountainous', residents: []}
    const wrapper = shallow(<Card card={card} />);
    expect(wrapper).toMatchSnapshot();
  })
})