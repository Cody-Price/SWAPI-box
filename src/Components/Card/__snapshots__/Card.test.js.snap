// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot if props includes "model" 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    card={
      Object {
        "model": "x-wing",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <h3 />
        <p>
          x-wing
        </p>
        <p />
        <p />
        <button
          className="fav-btn"
          onClick={[Function]}
        >
          Favorite
        </button>
      </div>,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3 />,
          <p>
            x-wing
          </p>,
          <p />,
          <p />,
          <button
            className="fav-btn"
            onClick={[Function]}
          >
            Favorite
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "x-wing",
          },
          "ref": null,
          "rendered": "x-wing",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorite",
            "className": "fav-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Favorite",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h3 />
          <p>
            x-wing
          </p>
          <p />
          <p />
          <button
            className="fav-btn"
            onClick={[Function]}
          >
            Favorite
          </button>
        </div>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3 />,
            <p>
              x-wing
            </p>,
            <p />,
            <p />,
            <button
              className="fav-btn"
              onClick={[Function]}
            >
              Favorite
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "x-wing",
            },
            "ref": null,
            "rendered": "x-wing",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Favorite",
              "className": "fav-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Favorite",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Card should match the snapshot if props includes "species" 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    card={
      Object {
        "species": "Human",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <h3 />
        <p />
        <p>
          Human
        </p>
        <p />
        <button
          className="fav-btn"
          onClick={[Function]}
        >
          Favorite
        </button>
      </div>,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3 />,
          <p />,
          <p>
            Human
          </p>,
          <p />,
          <button
            className="fav-btn"
            onClick={[Function]}
          >
            Favorite
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Human",
          },
          "ref": null,
          "rendered": "Human",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorite",
            "className": "fav-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Favorite",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h3 />
          <p />
          <p>
            Human
          </p>
          <p />
          <button
            className="fav-btn"
            onClick={[Function]}
          >
            Favorite
          </button>
        </div>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3 />,
            <p />,
            <p>
              Human
            </p>,
            <p />,
            <button
              className="fav-btn"
              onClick={[Function]}
            >
              Favorite
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Human",
            },
            "ref": null,
            "rendered": "Human",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Favorite",
              "className": "fav-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Favorite",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Card should match the snapshot if props includes "terrain" 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    card={
      Object {
        "residents": Array [],
        "terrain": "mountainous",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <h3 />
        <p />
        <p />
        <p>
          mountainous
        </p>
        <button
          className="fav-btn"
          onClick={[Function]}
        >
          Favorite
        </button>
      </div>,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3 />,
          <p />,
          <p />,
          <p>
            mountainous
          </p>,
          Array [],
          <button
            className="fav-btn"
            onClick={[Function]}
          >
            Favorite
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "mountainous",
          },
          "ref": null,
          "rendered": "mountainous",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorite",
            "className": "fav-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Favorite",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h3 />
          <p />
          <p />
          <p>
            mountainous
          </p>
          <button
            className="fav-btn"
            onClick={[Function]}
          >
            Favorite
          </button>
        </div>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3 />,
            <p />,
            <p />,
            <p>
              mountainous
            </p>,
            Array [],
            <button
              className="fav-btn"
              onClick={[Function]}
            >
              Favorite
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "mountainous",
            },
            "ref": null,
            "rendered": "mountainous",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Favorite",
              "className": "fav-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Favorite",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
